EventListener/TokenSubscriber.php

```diff
@@ -323,4 +325,94 @@ private function getCustomFieldValues(CustomObject $customObject, Token $token,
 
         return $fieldValues;
     }
+
+    public function onTokenReplacement(TokenReplacementEvent $event): void
+    {
+        $clickthrough = $event->getClickthrough();
+
+        if (!array_key_exists('dynamicContent', $clickthrough)) {
+            return;
+        }
+
+        $lead      = $event->getLead();
+        $tokenData = $clickthrough['dynamicContent'];
+        $tokens    = $clickthrough['tokens'];
+
+        foreach ($tokenData as $data) {
+            // Default content
+            $filterContent = $data['content'];
+
+            $isCustomObject = false;
+            foreach ($data['filters'] as $filter) {
+                foreach ($filter['filters'] as $condition) {
+                    if ('custom_object' !== $condition['object'] || empty($condition['display'])) {
+                        continue;
+                    }
+                    $isCustomObject = true;
+
+                    list($customObjectAlias, $customItemAlias) = array_map(
```

Comment:
I'm getting http `500` with `PHP Warning - Undefined array key 1` as the `$condition['display']` contains a string like `Animals Name`.

**We cannot base the determination of the custom object/item on `$condition['display']`.** It needs to be based on `$condition['field']` that contains either `cmf_$customObjectFieldId` or `cmo_$customObjectId` depending on the filter type (Custom object name or Custom object field).